### ISSUE TITLE
crashe in SSEStream when network is lost

### ISSUE THREAD

**Issue by dabitdev** [stellar/java-stellar-sdk#175]
#169  this was closed and the crashes were found in 0.4.1
nevermind, If I open a stream and the device is moved to airplane mode these following crash is always observed.

```
  Process: com.blockeq.stellarwallet, PID: 4665
    java.lang.IllegalStateException: Failed 200
        at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onFailure(SSEStream.java:176)
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:83)
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54)
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: javax.net.ssl.SSLException: Read error: ssl=0x71aac2d788: I/O error during system call, Connection timed out
        at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method)
        at com.android.org.conscrypt.NativeSsl.read(NativeSsl.java:399)
        at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:546)
        at shadow.okio.Okio$2.read(Okio.java:140)
        at shadow.okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
        at shadow.okio.RealBufferedSource.request(RealBufferedSource.java:68)
        at shadow.okio.RealBufferedSource.require(RealBufferedSource.java:61)
        at shadow.okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:304)
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.readChunkSize(Http1Codec.java:469)
        at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.read(Http1Codec.java:449)
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:394)
        at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:383)
        at shadow.okhttp3.internal.sse.ServerSentEventReader.processNextEvent(ServerSentEventReader.java:61)
        at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:80)
        at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54) 
        at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153) 
        at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by dabitdev** [comment]
it is totally normal, what is not normal is to crash my application process.

**Comment by dabitdev** [comment]
@bartekn would you like me to propose some changes in a PR?

**Comment by tomquisel** [comment]
@dabitdev please! That would be a huge help.

**Comment by sonhvp** [comment]
I’m having the same issue. How can handle `IllegalStateException ` when the network is lost. The exception is thrown inside the static inner class StellarEventSourceListener.

**Comment by ChrisLMills** [comment]
Hi

I am getting the same intermittent error (posted below). I have been informed on stellar.stackexchange.com that a fix is on its way:

https://stellar.stackexchange.com/questions/2412/java-sdk-transaction-listener-stream-connection-failure

`Fatal Exception: java.lang.IllegalStateException: Failed 200
       at org.stellar.sdk.requests.SSEStream$StellarEventSourceListener.onFailure(SSEStream.java:176)
       at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:83)
       at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54)
       at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
       at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

Caused by javax.net.ssl.SSLException: Read error: ssl=0xc8baee60: I/O error during system call, Software caused connection abort
       at com.android.org.conscrypt.NativeCrypto.SSL_read(NativeCrypto.java)
       at com.android.org.conscrypt.SslWrapper.read(SslWrapper.java:391)
       at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:567)
       at shadow.okio.Okio$2.read(Okio.java:140)
       at shadow.okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
       at shadow.okio.RealBufferedSource.request(RealBufferedSource.java:68)
       at shadow.okio.RealBufferedSource.require(RealBufferedSource.java:61)
       at shadow.okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:304)
       at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.readChunkSize(Http1Codec.java:469)
       at shadow.okhttp3.internal.http1.Http1Codec$ChunkedSource.read(Http1Codec.java:449)
       at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:394)
       at shadow.okio.RealBufferedSource.indexOfElement(RealBufferedSource.java:383)
       at shadow.okhttp3.internal.sse.ServerSentEventReader.processNextEvent(ServerSentEventReader.java:61)
       at shadow.okhttp3.internal.sse.RealEventSource.processResponse(RealEventSource.java:80)
       at shadow.okhttp3.internal.sse.RealEventSource.onResponse(RealEventSource.java:54)
       at shadow.okhttp3.RealCall$AsyncCall.execute(RealCall.java:153)
       at shadow.okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)`

**Comment by tomquisel** [comment]
Should be fixed by @tamirms 's change in v0.7.0
